js/src/frontend/Frontend2.cpp

```diff
@@ -65,7 +65,9 @@ bool Jsparagus::initScript(JSContext* cx, JS::Handle<JSScript*> script,
 
   uint32_t codeLength = jsparagus.bytecode.len;
   uint32_t noteLength =
-      roundUp(1 + jsparagus.bytecode.len, 4) - (1 + jsparagus.bytecode.len);
+      roundUp(sizeof(ImmutableScriptData::Flags) + jsparagus.bytecode.len + 1,
```

Comment:
can you add some comment that explains what this code is doing, including the reference for the structure layout?
so that it's clear why this expression contains `sizeof(ImmutableScriptData::Flags)` and `jsparagus.bytecode.len`.
(around https://searchfox.org/mozilla-central/rev/cfd1cc461f1efe0d66c2fdc17c024a203d5a2fd8/js/src/vm/JSScript.h#1616-1635 )